### PR TITLE
[NFC] Some minor code cleanups.

### PR DESCRIPTION
**Pull request by schweitzpgi**

### CODE REVIEW

File: flang/lib/Lower/ConvertExpr.cpp [hunk 1]
```diff
@@ -2011,14 +2006,42 @@ class ArrayExprLowering {
...
+
+  CC genProcRef(const Fortran::evaluate::ProcedureRef &procRef,
+                llvm::Optional<mlir::Type> retTy) {
+    auto loc = getLoc();
+    if (procRef.IsElemental()) {
+      if (const auto *intrin = procRef.proc().GetSpecificIntrinsic()) {
+        (void)intrin; // NB: LLVM_ATTRIBUTE_UNUSED doesn't suppress warning
+        TODO(loc, "elemental intrinsic");
+      }
+      if (ScalarExprLowering::isStatementFunctionCall(procRef))
```
Comment (jeanPerier) [inline flang/lib/Lower/ConvertExpr.cpp:2023]: Are elemental statement function really a thing ? (they may be, I am just surprised).

**Review by jeanPerier — APPROVED** [review]
Looks good to me, thanks for the clean-ups.

File: flang/lib/Lower/ConvertExpr.cpp (hunk 1 above)
Comment (tskeith) [reply flang/lib/Lower/ConvertExpr.cpp:2023]: I don't think that statement functions can be elemental procedures (though they can be pure procedures). So it's a bug that we don't catch them in semantics.

Comment (schweitzpgi) [reply flang/lib/Lower/ConvertExpr.cpp:2023]: Thanks, Tim. I'll just make this a hard error then.

**Pull request merged**